### PR TITLE
Fix Pandas PR CI Trigger

### PR DESCRIPTION
**Pull request by srilman**
## Changes included in this PR

Fix a problem with PR CI where it is still running Pandas tests even when no compiler or Pandas files are modified. Turns out it was a dumb mistake mixing up the if and else parts of an if expression.

In addition, I rewrote and disabled a couple more tests from PR CI, in particular a lot of duplicate Iceberg tests that would take ~20 minutes in total and the table column deletion pass tests that would take ~15 minutes.

Finally, I updated some of the test times and added any new tests to the log. In total, PR CI seems to take 30 minutes in total with all of these changes. I plan to make some more changes, but probably during my oncall next week.

## Testing strategy

Running PR CI.

## User facing changes

None

## Checklist
- [x] Pipelines passed before requesting review. To run CI you must include `[run CI]` in your commit message.
- [x] I am familiar with the [Contributing Guide](https://github.com/bodo-ai/Bodo/blob/main/CONTRIBUTING.md) 
- [x] I have installed + ran pre-commit hooks.

### CODE REVIEW

File: bodo/tests/utils.py [hunk 1]
```diff
@@ -3223,15 +3223,15 @@ def pytest_slow_unless_changed(features):
...
     if compiler_files_were_changed
-    else pytest.mark.pandas
+    else compose_decos((pytest.mark.slow, pytest.mark.pandas))
 )
 
 # This is for marking an entire test file
 # (pytestmark = pytest_pandas)
 pytest_pandas = [pytest.mark.pandas] + (
-    [pytest.mark.slow] if compiler_files_were_changed else []
+    [] if compiler_files_were_changed else [pytest.mark.slow]
```
Comment (srilman) [inline bodo/tests/utils.py:3234]: This was the main mistake why Pandas tests would run in PR CI

File: bodo/tests/test_table_column_del.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
 )
 from bodo.utils.utils import is_expr
 
+pytestmark = pytest.mark.slow
```
Comment (srilman) [inline bodo/tests/test_table_column_del.py:27]: These tests take ~15 minutes in total on the PR CI machines. Considering they are compiler tests that don't change often, I marked as slow.

File: bodo/tests/test_sql.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
 )
 from bodo.utils.testing import ensure_clean_mysql_psql_table
 
-pytestmark = pytest.mark.sql
+pytestmark = [pytest.mark.sql, pytest.mark.slow]
```
Comment (srilman) [inline bodo/tests/test_sql.py:27]: Similarly, this feature is mostly broken and not often updated so marked as slow. If you have any suggestions for other tests to mark as slow, let me know.

File: bodo/tests/utils.py [hunk 4]
```diff
@@ -3223,15 +3223,15 @@ def pytest_slow_unless_changed(features):
 # This is for use as a decorator for a single test function.
 # (@pytest_mark_pandas)
 pytest_mark_pandas = (
```
Comment (IsaacWarren) [inline bodo/tests/utils.py:3225]: This looks like it works now but what about a skipif? Might be more straightforward

File: bodo/tests/test_iceberg/test_stream_iceberg_write.py [hunk 5]
```diff
@@ -207,14 +207,7 @@ def check_table_comment(
             )
 
 
-@pytest.mark.timeout(1000)
-@pytest.mark.parametrize("column_comments", ["test_col_comments", None])
-@pytest.mark.parametrize("table_comments", ["test_tbl_comments", "", None])
-@pytest.mark.parametrize("table_properties", [True, False])
 def test_iceberg_write_basic(
```
Comment (srilman) [inline bodo/tests/test_iceberg/test_stream_iceberg_write.py:210]: So originally, this test would test Iceberg stream write on all types with all combinations of column comments, table comments, and table properties. That's completely unnecessary since comments and properties are not affected by the dataframe type. So I split it into two separate tests. That made these tests 15-20 minutes faster

File: buildscripts/github/test_dur_bodo.json
Comment (IsaacWarren) [inline buildscripts/github/test_dur_bodo.json:1]: Do you have any ideas for automating these changes?

**Review by IsaacWarren — APPROVED** [review]
Thanks Srinivas

File: bodo/tests/utils.py (hunk 4 above)
Comment (srilman) [reply bodo/tests/utils.py:3225]: Yeah probably would. My hope was to use a pytest extension like testmon that looks at coverage data to figure out what tests are affected and auto-enable / auto-skip

**Review by ehsantn — APPROVED** [review]
Thanks for the fixes!
